precompiles/services/src/lib.rs

```diff
@@ -151,14 +158,54 @@ where
 		};
 		let value = BalanceOf::<Runtime>::decode(&mut &value_bytes[..])
 			.map_err(|_| revert("Value is not a valid balance"))?;
+
+		let ttl_bytes = {
+			let mut ttl_bytes = [0u8; core::mem::size_of::<U256>()];
+			ttl.to_little_endian(&mut ttl_bytes);
+			ttl_bytes
+		};
+
+		let ttl = BlockNumberFor::<Runtime>::decode(&mut &ttl_bytes[..])
+			.map_err(|_| revert("Invalid ttl"))?;
+
+		let amount = {
+			let mut amount_bytes = [0u8; core::mem::size_of::<U256>()];
+			amount.to_little_endian(&mut amount_bytes);
+			BalanceOf::<Runtime>::decode(&mut &amount_bytes[..])
+				.map_err(|_| revert("Amount is not a valid balance"))?
+		};
+
+		const ZERO_ADDRESS: [u8; 20] = [0; 20];
+
+		let (payment_asset, amount) = match (payment_asset_id.as_u32(), payment_token_address.0 .0)
+		{
+			(0, ZERO_ADDRESS) => (Asset::Custom(0u32.into()), value),
+			(0, erc20_token) => {
+				if value != Default::default() {
+					return Err(revert("Value should be zero for ERC20 payment asset"));
```

Comment:
Let's standardize these error messages as constants so it's more organized.

Reply:
Good idea.

Reply:
Done [`3e60a59` (#845)](https://github.com/tangle-network/tangle/pull/845/commits/3e60a5937d7d82699110272b2317b09a0bbf2698)